opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp

```diff
@@ -183,10 +183,13 @@ updateNewton(const BVectorWell& dwells,
             const double dx_limited = sign * std::min(std::abs(dwells[seg][SPres]) * relaxation_factor, max_pressure_change);
             // some cases might have defaulted bhp constraint of 1 bar, we use a slightly smaller value as the bhp lower limit for Newton update
             // so that bhp constaint can be an active control when needed.
-            // TODO: we might need to distinguish SPres of the top segment from the SPres of the other segments, since only the top segment
-            // is involved in the well constraints/control checking
-            constexpr double spres_lower_limit = 1. * unit::barsa - 1. * unit::Pascal;
-            value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, spres_lower_limit);
+            if (seg == 0) { // top segment pressure is the bhp
+                constexpr double bhp_lower_limit = 1. * unit::barsa - 1. * unit::Pascal;
+                value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, bhp_lower_limit);
+            } else {
+                // enforcing 0. as the lower limit for the segment pressure of other segment
+                value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, 0.);
```

Comment:
we should discuss whether even keeping 0. as the lower limit. or removing this limit and give a warning message when segment pressure goes below 0.0. @steink